applications/serial_lte_modem/doc/slm_description.rst

```diff
@@ -0,0 +1,818 @@
+      Replace *example.com* with the host name or IPv4 address of a UDP server and *1234* with the corresponding port.
+      Then retrieve the returned data.
+
+      .. parsed-literal::
+         :class: highlight
+
+         **AT#XSENDTO="**\ *example.com*\ **",**\ *1234*\ **,1,"Test UDP"**
+         #XSENDTO: 8
+         OK
+         **AT#XRECVFROM**
+         PONG: Test UDP
+         #XRECVFROM: 1, 14
+         OK
+
+   #. Send hexadecimal data to a UDP server on a specified port.
+      Replace *example.com* with the host name or IPv4 address of a UDP server and *1234* with the corresponding port.
+      Then retrieve the returned data.
+
+      .. parsed-literal::
+         :class: highlight
+
+         **AT#XSENDTO="**\ *example.com*\ **",**\ *1234*\ **,0,"DEADBEEF"**
+         #XSENDTO: 4
+         OK
+         **AT#XRECVFROM**
+         504F4E473A20DEADBEEF
+         #XRECVFROM: 0, 20
+         OK
+
+   #. Close the socket.
+
+      .. parsed-literal::
+         :class: highlight
+
+         **AT#XSOCKET=0**
+         #XSOCKET: 0, closed
+         OK
+
+#. Test a UDP client with connection-based UDP:
+
+   a. Open a UDP socket and connect to a UDP server on a specified port.
+      Replace *example.com* with the host name or IPv4 address of a UDP server and *1234* with the corresponding port.
+
+      .. parsed-literal::
+         :class: highlight
+
+         **AT#XSOCKET=1,2,0**
+         #XSOCKET: 2, 2, 0, 17
+         OK
+
+         **AT#XCONNECT="**\ *example.com*\ **",**\ *1234*
+         #XCONNECT: 1
+         OK
+
+   #. Send plain text data to the UDP server and retrieve the returned data.
+
+      .. parsed-literal::
+         :class: highlight
+
+         **AT#XSEND=1,"Test UDP"**
+         #XSEND: 8
+         OK
+
+         **AT#XRECV**
+         PONG: Test UDP
+         #XRECV: 1, 14
+         OK
+
+   #. Close the socket.
+
+      .. parsed-literal::
+         :class: highlight
+
+         **AT#XSOCKET=0**
+         #XSOCKET: 0, closed
+         OK
+
+#. Test a connection-based UDP client with UDP proxy service (note that these commands are available only if :option:`CONFIG_SLM_UDP_PROXY` is defined):
+
+   a. Check the available values for the XUDPCLI command.
+
+      .. parsed-literal::
+         :class: highlight
+
+         **AT#XUDPCLI=?**
+         #XUDPCLI: (0, 1, 2),<url>,<port>,<sec_tag>
+         OK
+
+   #. Create a UDP client and connect to a server.
+      Replace *example.com* with the host name or IPv4 address of a UDP server and *1234* with the corresponding port.
+
+      .. parsed-literal::
+         :class: highlight
+
+         **AT#XUDPCLI=1,"**\ *example.com*\ **",**\ *1234*
+         #XUDPCLI: 2 connected
+         OK
+
+   #. Send plain text data to the UDP server and check the returned data.
+
+      .. parsed-literal::
+         :class: highlight
+
+         **AT#XUDPSEND=1,"Test UDP"**
+         #XUDPSEND: 8
+         OK
+         #XUDPRECV: 1, 14
```

Comment:
Is there a command missing here?

Reply:
No, this is an unsolicited event telling MCU that type and length of received data, which should be followed by the data itself